### PR TITLE
Improve truncation_side

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Largely taken from #859, but redone in order to avoid documentation issues since the
switch to MD. @NielsRogge you are co-authored, since it's mainly your PR.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -256,7 +257,9 @@ def run(self):
 
 extras["serving"] = deps_list("pydantic", "uvicorn", "fastapi", "starlette")
 extras["audio"] = deps_list("librosa", "pyctcdecode", "phonemizer")
-extras["speech"] = deps_list("torchaudio") + extras["audio"]  # `pip install ".[speech]"` is deprecated and `pip install ".[torch-speech]"` should be used instead
+extras["speech"] = (
+    deps_list("torchaudio") + extras["audio"]
+)  # `pip install ".[speech]"` is deprecated and `pip install ".[torch-speech]"` should be used instead
```
Comment (sgugger) [inline setup.py:262]: Put the comment above to avoid the ugly restyling like this.

File: tests/test_tokenization_common.py [hunk 2]
```diff
@@ -1398,6 +1398,65 @@ def test_right_and_left_padding(self):
...
+                # RIGHT PADDING - Check that it correctly pads when a maximum length is specified along with the padding flag set to True
+                truncation_size = 3
+                tokenizer.truncation_side = "right"
+                encoded_sequence = tokenizer.encode(sequence, add_special_tokens=False)
+                sequence_length = len(encoded_sequence)
+                truncated_sequence = tokenizer.encode(
+                    sequence, max_length=sequence_length - truncation_size, truncation=True, add_special_tokens=False
+                )
+                truncated_sequence_length = len(truncated_sequence)
+                assert sequence_length == truncated_sequence_length + truncation_size
```
Comment (sgugger) [inline tests/test_tokenization_common.py:1416]: Please use `self.assertEqual` in tests (here and all occurrences below)

**Review by sgugger — APPROVED** [review]
Thanks for working on this. I don't think you have properly put Niels as Co-author, there needs to be a special syntax Co-authored-by xxx in a commit description for GitHub to recognize it.

File: setup.py (hunk 1 above)
Comment (Narsil) [reply setup.py]: Done in `tokenizers` PR (and rebased since)

File: tests/test_tokenization_common.py (hunk 2 above)
Comment (Narsil) [reply tests/test_tokenization_common.py]: Done ! (There are still quite a bit of old `assert` in the rest of the file, should I do it in a separate PR ?

**Comment by Narsil** [comment]
> Thanks for working on this. I don't think you have properly put Niels as Co-author, there needs to be a special syntax Co-authored-by xxx in a commit description for GitHub to recognize it.

The email I used was wrong, I had to dive into other git commits to find the proper email.

File: tests/test_tokenization_common.py (hunk 2 above)
Comment (sgugger) [reply tests/test_tokenization_common.py]: If you have time, that would be great!

Comment (Narsil) [reply tests/test_tokenization_common.py]: https://github.com/huggingface/transformers/pull/15012

**Pull request merged**